docs/proposals/raw-string-literals.rst

```diff
@@ -0,0 +1,267 @@
+described unambiguously in a context free grammar and could become a common
+cause of errors when implementing JMESPath parsers.
+
+Parsing JSON literals has other complications as well. Here are the steps
+needed to currently parse a JSON literal value in JMESPath:
+
+1. When a ````` token is encountered, begin parsing a JSON literal.
+2. Collect each character between the opening ````` and closing `````
+   token, including any escaped ````` characters (i.e., ``\```) and store the
+   characters in a variable (let's call it ``$lexeme``).
+3. Copy the contents of ``$lexeme`` to a temporary value in which all leading
+   and trailing whitespace is removed. Let's call this ``$temp`` (this is
+   currently not documented but required in the
+   `JMESPath compliance tests <https://github.com/jmespath/jmespath.test/blob/c532a20e3bca635fb6ca248e5e955e1bd146a965/tests/syntax.json#L592-L606>`_).
+4. If ``$temp`` can be parsed as valid JSON, then use the parsed result as the
+   value for the literal token.
+5. If ``$temp`` cannot be parsed as valid JSON, then wrap the contents of
+   ``$lexeme`` in double quotes and parse the wrapped value as a JSON string,
+   making the following expressions equivalent: ```foo``` == ```"foo"```, and
+   ```[1, ]``` == ```"[1, ]"```.
+
+It is reasonable to assume that the most common use case for a JSON literal in
+a JMESPath expression is to provide a string value to a function argument or
+to provide a literal string value to a value in a multi-select list or
+multi-select hash. In order to make providing string values easier, it was
+decided that JMESPath should allow the quotes around the string to be elided.
+
+This proposal posits that allowing quotes to be elided when parsing JSON
+literals should be deprecated in favor of adding a proper string literal rule
+to JMESPath.
+
+
+Specification
+=============
+
+A raw string literal is value that begins and ends with a single quote, does
+not interpret escape characters, and may contain escaped single quotes to
+avoid delimiter collision.
+
+
+Examples
+--------
+
+Here are several examples of valid raw string literals and how they are
+parsed:
+
+- A basic raw string literal, parsed as ``foo bar``::
+
+      'foo bar'
+
+- An escaped single quote, parsed as ``foo'bar``::
+
+      'foo\'bar'
+
+- A raw string literal that contains new lines::
+
+      'foo
+      bar
+      baz!'
+
+  The above expression would be parsed as a string that contains new lines::
+
+      foo
+      baz
+      bar!
+
+- A raw string literal that contains escape characters,
+  parsed as ``foo\nbar``::
+
+       foo\nbar
+
+
+ABNF
+----
+
+The following ABNF grammar rules will be added, and is allowed anywhere an
+expression is allowed::
+
+    raw-string        = "'" *raw-string-char "'"
+    ; Any character other than "\" or "\'" or "\"
+    raw-string-char   = (%x20-21 / %x23-5B / %x5D-10FFFF) / raw-string-escape
```

Comment:
Did you mean single quote here?  `%x22`, the first omitted character, is a double quote.